src/Env.hs

```diff
@@ -170,6 +153,20 @@ emptyArgList = do
 
 ----------------------------------- Typechecker aux functions ------------------------------------------
 
+-- | Compute the transitive closure of list of pairs
```

Comment:
I would have written this using an infinite list created by applying `iterate` function with `closureAccum`, dropping elements a fixpoint is reached. This has the advantage that you don't have to manually write a recursion (which is slower).

Reply:
Also, is there any condition under which this function doesn't terminate? If so, would be good to document it (or test to avoid it)